pkg/operator/configobserver/apiserver/observe_cors.go

```diff
@@ -15,33 +15,40 @@ var clusterDefaultCORSAllowedOrigins = []string{
 	`//localhost(:|$)`,
 }
 
-// ObserveAdditionalCORSAllowedOrigins observes the additionalCORSAllowedOrigins field
-// of the APIServer resource and sets the corsAllowedOrigins field of observedConfig
-func ObserveAdditionalCORSAllowedOrigins(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
-	return innerObserveAdditionalCORSAllowedOrigins(genericListers, recorder, existingConfig, []string{"corsAllowedOrigins"})
+type corsObserver struct {
+	corsAllowedOriginsPath []string
 }
 
-// ObserveAdditionalCORSAllowedOriginsToArguments observes the additionalCORSAllowedOrigins field
-// of the APIServer resource and sets the cors-allowed-origins field in observedConfig.apiServerArguments
-func ObserveAdditionalCORSAllowedOriginsToArguments(genericListers configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
-	return innerObserveAdditionalCORSAllowedOrigins(genericListers, recorder, existingConfig, []string{"apiServerArguments", "cors-allowed-origins"})
+// NewCORSObserver creates a new observer for the additionalCORSAllowedOrigins,
```

Comment:
nit: we could say that it observers `APIServer.Spec.AdditionalCORSAllowedOrigins`